### PR TITLE
[FLINK-28355][python][e2e] Clean up python environment after e2e tests

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request will clean up python environment after e2e tests*

## Brief change log

  - *Clean up python environment after e2e tests*

## Verifying this change

This change added tests and can be verified as follows:

  - *Original tests*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/dev/lint-python.sh [hunk 1]
```diff
@@ -854,3 +863,8 @@ pushd "$FLINK_PYTHON_DIR" &> /dev/null
 if [ $skip_checks -eq 0 ]; then
     check_stage
 fi
+
+if [[ ${CLEAN_UP_FLAG} -eq 1 ]]; then
+    rm -rf ${CONDA_HOME}
+    rm -rf ${STAGE_FILE}
```
Comment (MartijnVisser) [inline flink-python/dev/lint-python.sh:869]: Should we also clean up the `flink-python/.tox` folder?

**Review by MartijnVisser — APPROVED** [review]
Had one question but thanks for the overall improvement! Will you als backport it to the other release branches?

File: flink-python/dev/lint-python.sh (hunk 1 above)
Comment (HuangXingBo) [reply flink-python/dev/lint-python.sh:869]: Yes. Good catch. Although the `.tox` directory is not generated during the `e2e` stage. The .tox folder is generated during the `flink-python` module test stage.

Comment (HuangXingBo) [reply flink-python/dev/lint-python.sh:869]: It looks like tirggering the tox test in `e2e` stage. I need to correct the script.

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
The failing test in CI run https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=37660&view=results was FLINK-26721 which is unrelated to this change. Merging this now.

@HuangXingBo Will you also create the backports towards `release-1.15` and `release-1.14` ?

**Comment by HuangXingBo** [comment]
@MartijnVisser I have cherry-pick the patch towards release-1.15 and release-1.14

**Pull request merged**